manifests/simplerule.pp

```diff
@@ -54,7 +54,7 @@
 #   Enable traffic counters for the matched traffic.
 define nftables::simplerule (
   Enum['present','absent'] $ensure = 'present',
-  Pattern[/^[-a-zA-Z0-9_]+$/] $rulename = $title,
+  Pattern[/^[a-zA-Z0-9_]+(-\d+)?$/] $rulename = $title,
```

Comment:
How about creating a `Nftables::Rulename`  type  ? That regex is probably elsewhere as well - not looked.

Should stop it diverging.

Reply:
The [regex used in `nftables::simplerule`](https://github.com/voxpupuli/puppet-nftables/blob/6a4ffeadf03a0e25d32963d12dcfa68a1a1595d0/manifests/simplerule.pp#L57) is a subset of [the one in `nftables::rule`](https://github.com/duritong/puppet-nftables/blob/master/manifests/rule.pp#L6) (the first part of the big regexp which represents the chain is injected by `simplerule` from the value of a parameter). I don't find a way to reuse a regex declared as another type in a `Pattern` type so we'd have anyway to have to different types to maintain in this case.

In pseudocode something like:

```
type Nftables::LogicalRuleName = Pattern[/^[a-zA-Z0-9_]+(-\d+)?$/]
type Nftables::RuleName = Pattern[/^[a-zA-Z0-9_]+-#Nftables::LogicalRuleName_WithoutAnchors$/]
```

Reply:
I usually start putting things into dedicated types, when either they become too long/complex or are used in multiple places.

Reply:
having it as a custom datatype also makes it easy to document it with puppet-strings and to test it. both are important for types with regular expressions, because they are hard to read

Reply:
I don't have a strong opinion. I'm happy to create two types (`Nftables::RuleName` and `Nftables::SimpleRuleName`, for instance). It won't help with having duplicate code but it might make reading and testing code easier, I guess. Would that be okay?

Reply:
@nbarrientos yes, please create those two